addons/payment_buckaroo/models/payment_transaction.py

```diff
@@ -0,0 +1,131 @@
+            'Brq_returnerror': urls.url_join(base_url, BuckarooController._exception_url),
+            'Brq_returnreject': urls.url_join(base_url, BuckarooController._reject_url),
+            'Brq_culture': (processing_values.get('partner_lang') or 'en_US').replace('_', '-'),
+            'add_returndata': buckaroo_tx_values.pop('return_url', '') or '',
+        })
+        buckaroo_tx_values['Brq_signature'] = self.acquirer_id._buckaroo_generate_digital_sign(
+            buckaroo_tx_values, incoming=False
+        )
+        return buckaroo_tx_values
+
+    @api.model
+    def _get_tx_from_feedback_data(self, provider, data):
+        if provider != 'buckaroo':
+            return super()._get_tx_from_feedback_data(provider, data)
+
+        normalized_data = _normalize_dataset(data)
+        reference = normalized_data.get('BRQ_INVOICENUMBER')
+        payment_id = normalized_data.get('BRQ_PAYMENT')
+        shasign = normalized_data.get('BRQ_SIGNATURE')
+        if not all(reference and payment_id and shasign):
+            raise ValidationError(
+                "Buckaroo: " + _(
+                    "Received data with missing reference (%(ref)s), payment_id (%(pay_id)), or "
+                    "shasign (%(sign))",
+                    ref=reference, pay_id=payment_id, sign=shasign
+                )
+            )
+
+        tx = self.search([('reference', '=', reference)])
+        if not tx:
+            raise ValidationError(
+                "Buckaroo: " + _(
+                    "Received data with reference %s matching no transaction", reference
+                )
+            )
+
+        # Verify signature
+        shasign_check = tx.acquirer_id._buckaroo_generate_digital_sign(data, incoming=True)
+        if shasign_check.upper() != shasign.upper():
+            raise ValidationError(
+                "Buckaroo: " + _(
+                    "Invalid shasign: received %(sign)s, computed %(check)s",
+                    sign=shasign, check=shasign_check
+                )
+            )
+
+        return tx
+
+    def _process_feedback_data(self, data):
+        self.ensure_one()
+        if self.provider != 'buckaroo':
+            return super()._process_feedback_data(data)
+
+        normalized_data = _normalize_dataset(data)
+        if float_compare(float(normalized_data.get('BRQ_AMOUNT', '0.0')), self.amount, 2) != 0:
+            raise ValidationError("Buckaroo: " + _("The paid amount does not match the total."))
+        if normalized_data.get('BRQ_CURRENCY') != self.currency_id.name:
+            raise ValidationError(
+                "Buckaroo: " + _(
+                    "The currency returned by Buckaroo %(rc)s does not match the transaction "
+                    "currency %(tc)s.",
+                    rc=normalized_data.get('BRQ_CURRENCY'), tc=self.currency_id.name
+                )
+            )
+
+        self.acquirer_reference = normalized_data.get('BRQ_TRANSACTIONS')
```

Comment:
What if there is more than one transaction?

As per the doc:
> One or more unique keys referring to transactions which are linked to
this payment. Multiple keys are separated by a comma (,)